src/lib/stores/bluetoothRead.ts

```diff
@@ -0,0 +1,145 @@
+import { everyEqual } from "$lib/array/everyEqual";
+import {
+  configurationPayloadTrailer,
+  radarDataOutputPayloadTrailer,
+} from "$lib/ld2410/constants";
+
+type ReadResult =
+  | { value: Uint8Array; done: false }
+  | { value: undefined; done: true };
+
+interface ReadEvent {
+  eventType: "READ";
+  payload: Uint8Array;
+}
+
+interface WriteEvent {
+  eventType: "WRITE";
+  payload: Uint8Array;
+}
+
+interface ConnectEvent {
+  eventType: "CONNECT";
+}
+
+interface SubscribedEvent {
+  eventType: "SUBSCRIBED";
+}
+
+interface DisconnectEvent {
+  eventType: "DISCONNECT";
+}
+
+type SerialEvent =
+  | ReadEvent
+  | WriteEvent
+  | ConnectEvent
+  | DisconnectEvent
+  | SubscribedEvent;
+
+type SubscribeCallback = (value: SerialEvent) => void;
+
+interface Store {
+  subscribe: (subscription: SubscribeCallback) => () => void;
+}
+
+export type SerialStore = Store & {
+  connect: () => void;
+  disconnect: () => void;
+  write: (payload: Uint8Array) => void;
+};
+
+export const createBluetoothReadStore = (
+  server: any
+): SerialStore => {
+  let stopping = false;
+  let sendPassCharacteristic: any | null = null;
+  let readDataCharacteristic: any | null = null;
```

Comment:
With the types library I suggested above:

```suggestion
  server: BluetoothRemoteGATTServer
): SerialStore => {
  let stopping = false;
  let sendPassCharacteristic: BluetoothRemoteGATTCharacteristic | null = null;
  let readDataCharacteristic: BluetoothRemoteGATTCharacteristic | null = null;
```

and go from there...